### PR TITLE
Continuing top-down revisions of fuzzy storage worker doc

### PR DESCRIPTION
**Pull request by TonyGravagno**
Highlights:

- Added TOC
- Re-organized a few topics for consistency
- Many updates to wording and detail

Commits are mostly presented as a series of top-down changes, with only one or two sections per change. View in time-order if required to see how it progresses.

This is more of a transitional PR, though the doc is in good shape. The next PR should complete this series of changes for this page, and the final PR should just have final tweaks.

### CODE REVIEW

File: doc/fuzzy_storage.md [hunk 1]
```diff
@@ -130,27 +133,67 @@ worker "fuzzy" {
...
+It would be prudent to compare the volume of hashes learned over some time with available RAM. For example, 400 thousands hashes may occupy about 100 Mb and 1.5 million hashes may occupy 0.5 Gb. It is not recommended to increase storage size more than the available RAM size due to a significant performance degradation. That is, don't rely on swap space, and don't choke other processes for resources. If you have a small volume of hashes suitable for learning, start with an expiration time of 90 days. Tune that down if the volume of data over that time period results in an unsuitable amount of available RAM - for example, if peak-time available RAM goes down to 20%, reduce the expiration time to 70 days, and see if data expiring from storage releases a more acceptable amount of RAM.
+
+
+### Access control
+
+Rspamd does not allow changes to fuzzy storage by default. Any server connecting via UDP to the fuzzy_storage process must be authorized. A list of trusted IP-addresses and/or networks must be provided to make learning possible. Practically, it is better to write from the local address only (127.0.0.1) since fuzzy storage uses UDP that is not protected from source IP forgery.
```
Comment (vstakhov) [inline doc/fuzzy_storage.md:167]: From the perspective of the `fuzzy_storage` they are clients, not servers.

File: doc/fuzzy_storage.md [hunk 2]
```diff
@@ -130,27 +133,67 @@ worker "fuzzy" {
...
   allow_update = ["127.0.0.1"];
 
   # or 10.0.0.0/8, for internal network
 }
 ~~~
 
-Transport encryption might also be used for access control purposes.
+The `allow_update` setting is a comma-delimited array of strings, or a [map]({{ site.baseurl }}/doc/modules/multimap.html) of IP addresses, that are allowed to perform changes to fuzzy storage - You should also set `read_only` = no in your fuzzy_check plugin, see step 3 below.
+
+Transport protocol encryption might also be used for access control purposes...
```
Comment (vstakhov) [inline doc/fuzzy_storage.md:180]: I'd say that so far transport encryption can only distinguish between the states access/no-access but not between the states like read-only/read-write. However, it is very powerful for the first purpose and might be useful for the second one (once implemented).

File: doc/fuzzy_storage.md [hunk 3]
```diff
@@ -163,9 +206,9 @@ keypair {
 }
 ~~~
 
-This command creates a **unique**  keypair, where **public** key should be copied manually to the customer's host (e.g. via ssh) or published in any way to guarantee the reliability (e.g. certified digital signature or HTTPS-site hosting).
+The  **public** `pubkey` should be copied manually to the remote host, or published in any way that guarantees the reliability (e.g. certified digital signature or HTTPS-site hosting). As always the **private** `privkey` should never be published or shared.
 
-Each storage can use any number of keys simultaneously:
+Each storage can use any number of keys simultaneously, one for each remote client:
```
Comment (vstakhov) [inline doc/fuzzy_storage.md:211]:
```suggestion
Each storage can use any number of keys simultaneously, one for each remote client (or a group of clients):
```

File: doc/fuzzy_storage.md [hunk 4]
```diff
@@ -185,11 +228,7 @@ worker "fuzzy" {
 }
 ~~~
 
-This feature is useful for creating restricted storages where access is allowed merely to those customers who knows about one of the public keys of storage:
-
-<center><img class="img-responsive" src="{{ site.baseurl }}/img/rspamd-fuzzy-3.png" width="75%"></center>
-
-To enable such a mandatory encryption mode you should use `encrypted_only` option:
+This mechanism is optional unless explicitly made mandatory. To enable mandatory encryption mode, add the `encrypted_only` option. Then, client systems without a valid public key are not able to access the storage in this mode.
```
Comment (vstakhov) [inline doc/fuzzy_storage.md:231]:
```suggestion
This mechanism is optional unless explicitly made mandatory. To enable mandatory encryption mode, add the `encrypted_only` option. In this mode, client systems that do not use a valid public key will not able to access the storage.
```

**Review by vstakhov — APPROVED** [review]

File: doc/fuzzy_storage.md (hunk 1 above)
Comment (TonyGravagno) [reply doc/fuzzy_storage.md]: Good catch, fixed.

File: doc/fuzzy_storage.md (hunk 2 above)
Comment (TonyGravagno) [reply doc/fuzzy_storage.md]: Line removed.

**Comment by TonyGravagno** [comment]
Ready for another round. Thanks!

**Pull request merged**